configurate-core/src/test/java/ninja/leaping/configurate/objectmapping/TypeSerializersTest.java

```diff
@@ -255,6 +258,31 @@ public void testMapSerializerPreservesEmptyMap() throws ObjectMappingException {
     private static class TestObject {
         @Setting("int") private int value;
         @Setting private String name;
+
```

Comment:
I'd like to see "sad path" testing too - there are two scenarioes that you've introduced that should fail in an expected way:

* The `TypeSerializer` has not got a public, no-args constructor
* The `TypeSerializer` deserialises the value into a type that does not fit into the destination field (`ClassCastException`).

In both cases, a specific `ObjectMappingException` with a specific inner exception. Can we ensure that the expected messages are thrown? That way, we can ensure that users of this API know why they went wrong.